### PR TITLE
made context manager for set

### LINKED ISSUE #1117: Settable parameters as context managers
It might be useful for parameters which can be set to act as context managers, such that user code can be guaranteed that a parameter is returned to its original value after a block of code. For instance:

```python
>>> v = dac.voltage()
>>> with dac.voltage(-1):
...     # Do stuff with the DAC output set to -1 V.
...
>>> assert abs(dac.voltage() - v) <= tolerance
```
@QCoDeS/core

### PR DESCRIPTION
**Pull request by sohailc**
Fixes #1117.

### CODE REVIEW

File: qcodes/tests/test_parameter.py [hunk 1]
```diff
@@ -902,3 +902,31 @@ def tearDown(self):
...
+        self.instrument.close()
+        del self.instrument
+
+    def test_none_value(self):
+        with self.instrument.a.set(3):
+            assert self.instrument.a.get() == 3
+        assert self.instrument.a.get() is None
+
+    def test_context(self):
+        self.instrument.a.set(1)
```
Comment (astafan8) [inline qcodes/tests/test_parameter.py]: what is this line needed for?

Comment (sohailc) [reply qcodes/tests/test_parameter.py]: forgot to remove it

**Comment by sohailc** [comment]
@jenshnielsen can you approve this now?

**Comment by jenshnielsen** [comment]
Seems sensible to me. I am still in doubt about is this should bind directly to set. @WilliamHPNielsen any opinion

**Comment by WilliamHPNielsen** [comment]
I would probably prefer to bind this to something called `set_to`, simply because I conjecture that users will find that (slightly) easier to understand. And I also personally find it nice that the line
```
with dac.voltage.set_to(0):
```
kind of reads out like a natural language command.

But I don't feel very strongly about either option.

**Comment by astafan8** [comment]
shall we merge this pull-request or is there anything to do?

**Comment by astafan8** [comment]
Let's poll for the name of the method that can be used as context manager for the "set" functionality:
/polls "existing set() - with dac.voltage.set(1):" "set_to() - with dac.voltage.set_to(1):"

**Comment by sohailc** [comment]
@jenshnielsen @astafan8 @WilliamHPNielsen  Can we have another look, please?

**Comment by Dominik-Vogel** [comment]
Sorry for the delay, I was out last week.
I find the two naming options slightly confusing. For an unexperienced user a `with` statement is something exotic and the prose english interpretation of the two suggested names does not reflect its behavior.
I would prefer something like
```
with dac.voltage.temporarily_at(0)
```

**Pull request merged**